overlord/snapstate/storehelpers.go

```diff
@@ -629,9 +629,21 @@ func refreshCandidates(ctx context.Context, st *state.State, names []string, use
 	ignoreValidationByInstanceName := make(map[string]bool)
 	nCands := 0
 
-	enforcedSets, err := EnforcedValidationSets(st)
-	if err != nil {
-		return nil, nil, nil, err
+	var enforcedSets *snapasserts.ValidationSets
+	var revOptsByName map[string]*RevisionOptions
+
+	// if refreshing to specific revision to enforce a new validation set, we've
```

Comment:
I'm not entirely sure I understand this comment. Does that mean, that in the event we have a validation set enforced, which is valid, and we want to refresh to another revision of that snap. That once we are here, the revision we are refreshing to is also valid?

Or are we ignoring validation sets if we are refreshing to a specific revision?

Reply:
We ignore validation sets here if we're refreshing to specific revision. In this case, we've (or should have at least) already checked it against currently enforced sets